cpp/include/cudf/io/parquet.hpp

```diff
@@ -846,6 +846,18 @@ class chunked_parquet_writer_options_builder {
     return *this;
   }
 
+  /**
+   * @brief Sets decimal precision data.
+   *
+   * @param v Vector of precision data flattened with exactly one entry per
+   *          decimal column.
+   */
+   chunked_parquet_writer_options_builder& decimal_precision(std::vector<uint8_t> const& v)
```

Comment:
How is this intended to work with nested types, like a list of decimal values? or a list of struct of decimal to long?

Also is a `uint8_t` really the maximum precision that parquet can support? I know it is above the maximum that we can support right now, and can handle 128-bit support too, but it would be good to know if parquet can support higher, jsut to future proof the API.

Reply:
> How is this intended to work with nested types, like a list of decimal values? or a list of struct of decimal to long?

I don't know if I am understanding your question correctly. Its a flattened list of precisions so a list of decimal types will only have one precision, @hyperbolic2346 can you correct me if I am wrong? 

> Also is a uint8_t really the maximum precision that parquet can support? I know it is above the maximum that we can support right now, and can handle 128-bit support too, but it would be good to know if parquet can support higher, jsut to future proof the API.

So parquet is capable of handling decimals backed by `fixed_len_byte_array` in which case the precision is limited by the size of the array. Since this is just a small wrapper around the underlying cudf code we can't change it right now.

Reply:
`uint8_t` was selected in cudf because it seemed like there would never be a need for higher precision. I'm happy to adjust that if the need is there. I didn't consider the case of a `fixed_len_byte_array` since it isn't exactly supported in cudf currently.

The decimal precision array is a flattened array of values, meaning that if you have 4 decimal columns and 1 struct column with 2 decimals inside the array will need 6 values total. Each time we write out a column metadata for a decimal, we pop a value off the precision array and use that. The code will throw a logic error if too many or too few entries are provided.